src/sdk/pynni/nni/compression/torch/builtin_pruners.py

```diff
@@ -257,22 +256,16 @@ def _get_distance_sum(self, weight, in_idx, out_idx):
         out_idx: int
             output channel index of specified filter, this method calculates the total distance
             between this specified filter and all other filters.
-        in_idx: int
-            input channel index of specified filter
         Returns
         -------
         float32
             The total distance
         """
         logger.debug('weight size: %s', weight.size())
-        if len(weight.size()) == 4:  # Conv2d
-            w = weight.view(-1, weight.size(-2), weight.size(-1))
-            anchor_w = weight[out_idx, in_idx].unsqueeze(0).expand(w.size(0), w.size(1), w.size(2))
-        elif len(weight.size()) == 3:  # Conv1d
-            w = weight.view(-1, weight.size(-1))
-            anchor_w = weight[out_idx, in_idx].unsqueeze(0).expand(w.size(0), w.size(1))
-        else:
-            raise RuntimeError('unsupported layer type')
+        assert len(weight.size()) in [3, 4], 'unsupported weight shape'
+
+        w = weight.view(weight.size(0), -1)
+        anchor_w = w[out_idx].unsqueeze(0).expand(w.size(0), w.size(1))
         x = w - anchor_w
         x = (x * x).sum((-2, -1))
```

Comment:
why here is `x = (x * x).sum((-2, -1))`? x should be a matrix with shape `c_out * (c_in * k * k)`, to get the distance sum, maybe it should be `x = (x * x).sum(-1)` and then sqrt and sum.

Reply:
Thanks, I forgot to update here after reshape.

Reply:
It still can pass UT because for the UT case, the mistake changed the value of distance, but the sort order of distances are still correct.   We can see that the UT case are not strong enough.  Should be improved later.